### PR TITLE
Define a XMLHttpRequestBodyInit without ReadableStream

### PR DESCRIPTION
**Pull request by gterzian**
Part of https://github.com/whatwg/xhr/issues/277, removing `ReadableStream` from the body init used by XHR, and re-defining a `BodyInit` to include the stream option.

- [x] At least two implementers are interested (and none opposed):
   * Mozilla cc @annevk
   * Google cc @yutakahirano
- [x] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   * I think this should not affect existing tests in fetch, for xhr see https://github.com/whatwg/xhr/issues/277
- [ ] [Implementation bugs](https://github.com/whatwg/meta/blob/master/MAINTAINERS.md#handling-pull-requests) are filed:
   * None are filed because the changes here described the current behavior.

(See [WHATWG Working Mode: Changes](https://whatwg.org/working-mode#changes) for more details.)

***

### :boom: Error: 400 Bad Request :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Jun 2, 2020, 4:27 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [CSS Spec Preprocessor](https://api.csswg.org/bikeshed/) - CSS Spec Preprocessor is the web service used to build Bikeshed specs.

:link: [Related URL](https://api.csswg.org/bikeshed/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fwhatwg%2Ffetch%2Fe057e92ab68b940580ab3302a46b3e91317e0652%2Ffetch.bs&force=1&md-status=LS-PR&md-Text-Macro=PR-NUMBER%201029)

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20whatwg/fetch%231029.)._
</details>

### CODE REVIEW

**Comment by gterzian** [comment]
This one seems to build locally, is the error above indicative of a problem?

**Comment by yutakahirano** [comment]
Can you file a bug for each browser? You can refer to the same bugs here and https://github.com/whatwg/xhr/pull/279.

**Comment by yutakahirano** [comment]
Ah sorry, no bugs are needed because this is specifying the current behavior. Can you state that in the description?

**Comment by gterzian** [comment]
Ok, I have added

> None are filed because the changes here described the current behavior.

Reading `MAINTAINERS.md`, I think this might impact docs, MDN explicitly states a `ReadableStream` is an option, so I'll cc @whatwg/documentation and add the label.

**Comment by CYBAI** [comment]
> Also it appears @whatwg/documentation is not an actual team?

IIRC, GitHub will only make the text mention-able for a team member's mention so maybe you will need someone (a team member of `@whatwg/*`)'s help to do the mention 👀

**Comment by gterzian** [comment]
I have added the labels and cc'ed the team on the companion PR in XHR, as I think it's that one affecting docs rather than this one...

@annevk thank you for adding me to the org!

File: fetch.bs [hunk 1]
```diff
@@ -5458,7 +5458,10 @@ method, when invoked, must run these steps:
 <h3 id=body-mixin>Body mixin</h3>
 
 <pre class=idl>
-typedef (Blob or BufferSource or FormData or URLSearchParams or ReadableStream or USVString) BodyInit;</pre>
+typedef (Blob or BufferSource or FormData or URLSearchParams or USVString) XMLHttpRequestBodyInit;</pre>
+
+<pre class=idl>
```
Comment (annevk) [inline fetch.bs:5463]:
```suggestion
```

**Comment by yutakahirano** [comment]
cc: @yoichio fyi

File: fetch.bs [hunk 2]
```diff
@@ -5458,7 +5458,9 @@ method, when invoked, must run these steps:
 <h3 id=body-mixin>Body mixin</h3>
 
 <pre class=idl>
-typedef (Blob or BufferSource or FormData or URLSearchParams or ReadableStream or USVString) BodyInit;</pre>
+typedef (Blob or BufferSource or FormData or URLSearchParams or USVString) XMLHttpRequestBodyInit;</pre>
+
+typedef (ReadableStream or XMLHttpRequestBodyInit) BodyInit;</pre>
```
Comment (annevk) [inline fetch.bs:5463]:
```suggestion
typedef (Blob or BufferSource or FormData or URLSearchParams or USVString) XMLHttpRequestBodyInit;

typedef (ReadableStream or XMLHttpRequestBodyInit) BodyInit;
</pre>
```

File: fetch.bs [hunk 3]
```diff
@@ -5458,7 +5458,10 @@ method, when invoked, must run these steps:
 <h3 id=body-mixin>Body mixin</h3>
 
 <pre class=idl>
-typedef (Blob or BufferSource or FormData or URLSearchParams or ReadableStream or USVString) BodyInit;</pre>
+typedef (Blob or BufferSource or FormData or URLSearchParams or USVString) XMLHttpRequestBodyInit;
+
+typedef (ReadableStream or XMLHttpRequestBodyInit) BodyInit;
+</pre>
```
Comment (annevk) [inline fetch.bs:5464]:
```suggestion
typedef (ReadableStream or XMLHttpRequestBodyInit) BodyInit;</pre>
```

**Pull request merged**